### PR TITLE
Add Snake

### PR DESCRIPTION
**Pull request by x-sheep**
This puzzle uses the presentation used by Toketa, as discussed in #136.

### CODE REVIEW

File: src/variety/snake.js [hunk 1]
```diff
@@ -0,0 +1,409 @@
...
+
+	Encode: {
+		decodePzpr: function(type) {
+			this.decodeCircle();
+			this.outbstr = this.outbstr.substr(1);
+			this.decodeNumber16ExCell();
+		},
+		encodePzpr: function(type) {
+			this.encodeCircle();
+			this.outbstr += "/";
```
Comment (robx) [inline src/variety/snake.js:197]: Any particular reason to add this separator? Other types don't tend to do that. (If we were to start something like this, the decoder should make use of it, i.e., split on `/` and then decode both parts separately.)

Comment (x-sheep) [reply src/variety/snake.js]: Aquarium needed the extra slash (for reasons I can't recall), but Snake apparently doesn't. I've removed it here.

**Comment by robx** [comment]
Gave this a test now, and I think we can merge. Some of the errors feel a bit awkward to me, but I don't have particularly good ideas for improving things. Maybe you have an idea?

1. "The snake branches off". The following error confused me quite a bit. Two parts to this: (a) that the error highlights only the center, not the whole T (though changing that really wouldn't help in this situation (b) to me this particular situation is rather the snake doubling back on itself than branching.
Screenshot 2020-04-28 at 19 41 27

2. "The snake has no head or tail". Feels like in this scenario, the problem isn't really the lack of head or tail, but rather that the snake is divided. Would changing the order of the checks help? But even when it's just a single 2x2 square, the lack of head or tail is again not what I first think of.
Screenshot 2020-04-28 at 19 41 47

Perhaps both together could be addressed by adding a dedicated no-2x2 check that says something like "The snake touches itself" or "The snake doubles back on itself"? What do you think? (This is not about correctness, just about providing more "intuitive" error messages.)

**Comment by x-sheep** [comment]
I'm not sure. Maybe it's a good idea to deploy this to a separate environment and ask others for feedback?

**Comment by tausbn** [comment]
I like the idea of having a separate 2x2 check with "the snake loops back on itself" or whatever as the error message. (I would actually be fine with the error message stating 2x2 squares are disallowed, even if this is only an "implied" constraint. I understand why some people might not want this, however.)

And of course this check would have to precede the branching check for maximum effectiveness.

**Comment by x-sheep** [comment]
I've added a 2x2 check, and moved the check for loops below the connectedness check.

**Comment by robx** [comment]
This feels right, thanks! Merged by hand.

**Pull request closed without merging**